services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -195,6 +195,35 @@ const indexBlock = async job => {
 					}, dbTrx);
 				}
 			}
+
+			// TODO: Verify and enable it once pos:validatorStaked schema is exposed from SDK
+			// const tokenIDLockedAmountChangeMap = {};
+			// events.forEach(event => {
+			// 	const { data: eventData } = event;
+			// 	// Initialize map entry with BigInt
+			// 	if (['locked', 'unlocked'].includes(event.name) &&
+			//		!(eventData.tokenID in tokenIDLockedAmountChangeMap[eventData.tokenID])) {
+			// 		tokenIDLockedAmountChangeMap[eventData.tokenID] = BigInt(0);
+			// 	}
+
+			// 	if (event.name === 'locked') {
+			// 		tokenIDLockedAmountChangeMap[eventData.tokenID] += eventData.amount;
+			// 	} else if (event.name === 'unlocked') {
+			// 		tokenIDLockedAmountChangeMap[eventData.tokenID] -= eventData.amount;
+			// 	}
+			// });
+
+			// Object.entries(tokenIDLockedAmountChangeMap).forEach(([tokenID, lockedAmountChange]) => {
+			// 	const tokenKey = `${keyValueTable.KEYS.TOTAL_LOCKED_PREFIX}_${tokenID}`;
+			// 	const curLockedAmount = BigInt(await keyValueTable.get(tokenKey) || 0);
+			// 	const newLockedAmount = curLockedAmount + lockedAmountChange;
+
+			// 	if (newLockedAmount === 0) {
+			// 		keyValueTable.delete(tokenKey);
+			// 	} else {
+			// 		keyValueTable.set(tokenKey, newLockedAmount);
+			// 	}
```

Comment:
Please ensure that these operations are part of the same DB transaction of which the block updates also are.
Also, `await` the updates.

`newLockedAmount === 0` will always evaluate to `false`. Kindly adjust accordingly. In fact, do not delete the entry.

```suggestion
			// 	await keyValueTable.set(tokenKey, newLockedAmount, dbTrx);
```

Reply:
Applied.